docs/using-the-aws-driver/OktaAuthentication.md

```diff
@@ -0,0 +1,51 @@
+### Okta Authentication Plu# Okta Authentication Plugin
+
+The Okta Authentication Plugin adds support for authentication via Federated Identity and then database access via IAM.
+
+## What is Federated Identity
+
+Federated Identity allows users to use the same set of credentials to access multiple services or resources across
+different organizations. This works by having Identity Providers (IdP) that manage and authenticate user credentials,
+and Service Providers (SP) that are services or resources that can be internal, external, and/or belonging to various
+organizations. Multiple SPs can establish trust relationships with a single IdP.
+
+When a user wants access to a resource, it authenticates with the IdP. From this an authentication token generated and is
+passed to the SP then grants access to said resource.
+In the case of AD FS, the user signs into the AD FS sign in page. This generates a SAML Assertion which acts as a
+security token. The user then passes the SAML Assertion to the SP when requesting access to resources. The SP verifies
+the SAML Assertion and grants access to the user.
```

Comment:
Any reason why we are manually adding line breaks instead of using word wrap? This renders fine on GitHub but depending on the renderer, it might look worse with the manual line breaks.

e.g.
![image](https://github.com/user-attachments/assets/ead70300-769b-4267-af1d-2874a9fcb382)

Reply:
Auto-formatting messed it up.